client/scripts/bundle.js

```diff
@@ -182,39 +182,48 @@ async function main() {
     process.exit(1);
   }
 
+  console.log("> Fetching external server dependencies...")
+  const serverPackageJson = JSON.parse(fs.readFileSync(path.join("..", "server", "package.json"), "utf8"));
```

Comment:
nit: in case cwd changes:

```suggestion
  const serverPackageJson = JSON.parse(fs.readFileSync(path.join(__dirname, "..", "server", "package.json"), "utf8"));
```

Reply:
Done in https://github.com/NomicFoundation/hardhat-vscode/pull/545/commits/798fbd11b5b8eb7fbd2b1591f09d564bb91e6230